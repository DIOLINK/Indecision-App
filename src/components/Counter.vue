<template>
  <h2>{{customTitle}}</h2>
  <p><span data-testid="counter" >{{counter}}</span> <sup>2</sup> = {{squareCounter}} </p>
  <div>
    <button data-testid="increment" @click="increment()">+1</button>
    <button data-testid="decrement" @click="decrement()">-1</button>
  </div>
</template>

<script>
export default {
  props:{title: String, start:{
    type:Number,
    default:100, 
    validator: function(value){
      return value>=100;
    }},
  },
  data() {
    return {
      counter: this.start,
    }
  },
  methods:{
    getSquareValue() {
     return this.counter * this.counter;
    },
    increment(){
        this.counter++
    },
    decrement(){  
        this.counter--
    },
  },
  computed: {
    squareCounter() {
       return this.counter * this.counter
    },
    customTitle() {
      return this.title || 'Counter'
    }
  },
}
</script>

<style>
button{
  background-color: #64B687;
  border-radius: 6px;
  color: white;
  cursor: pointer;
  margin: 0 5px;
  padding: 5px 10px;
  transition: 0.3s ease-in-out;

}
</style>