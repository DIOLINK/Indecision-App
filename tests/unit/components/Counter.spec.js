import Counter from '@/components/Counter';
import { shallowMount } from '@vue/test-utils';
describe('Test in Counter component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(Counter);
  });

  it('Debe de hacer match con el snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  it('Shot render h2', () => {
    expect(wrapper.find('h2').text()).toBe('Counter');
  });
  it('Debe de incrementar en 1 y decrementar en 2', async () => {
    const [increment, decrement] = wrapper.findAll('button');
    const counter = wrapper.find('[data-testid="counter"]');
    await increment.trigger('click'); //101

    console.log('counter :>> ', counter.text());
    expect(counter.text()).toBe('101');

    await decrement.trigger('click'); //100
    await decrement.trigger('click'); //99
    console.log('counter :>> ', counter.text());
    expect(counter.text()).toBe('99');
  });
  it('Debe de identificar si las props son correctas', () => {
    console.log('start :>> ', wrapper.props());
    const { start, title } = wrapper.props();
    const counter = wrapper.find('[data-testid="counter"]');
    expect(Number(counter.text())).toBe(start);

    expect(title).toBe(undefined);
  });
  it('Debe de mostrar la prop title', () => {
    const wrapper = shallowMount(Counter, {
      props: {
        title: 'Counter 100',
      },
    });
    expect(wrapper.find('h2').text()).toBe('Counter 100');
  });
});
