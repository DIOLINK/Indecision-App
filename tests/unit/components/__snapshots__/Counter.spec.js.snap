// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in Counter component Debe de hacer match con el snapshot 1`] = `
<h2>Counter</h2>
<p><span data-testid="counter">100</span> <sup>2</sup> = 10000</p>
<div><button data-testid="increment">+1</button><button data-testid="decrement">-1</button></div>
`;
